File: frontend/src/shared/types/Prescription.ts
interface IObjectKeys {
  [key: string]: string | number | null;
}
export interface Prescription extends IObjectKeys {
  id: string;
  medication: string;
  dosage: string;
  notes: string;
  refills: number;
  started: string | null;
  ended: string | null;
}

export const generatePrescriptionTemplate = () => {
  return {
    id: "",
    medication: "",
    dosage: "",
    notes: "",
    refills: 0,
    started: "",
    ended: "",
  };
};

File: backend/mtg/internal/models/dto/prescription/prescription.go
package dto

import "time"

type PrescriptionDTO struct {
	Medication string     `json:"medication"`
	Dosage     string     `json:"dosage"`
	Notes      *string    `json:"notes"`
	Started    *time.Time `json:"started"`
	Ended      *time.Time `json:"ended"`
	Refills    *int       `json:"refills"`
	Total      int        `json:"total"`
	Owner      *string    `json:"owner"`
}
